metals/src/main/scala/scala/meta/internal/debug/JvmDebugAdapter.scala

```diff
@@ -0,0 +1,53 @@
+package scala.meta.internal.debug
+
+import java.util.concurrent.CompletableFuture
+import org.eclipse.lsp4j.debug
+import org.eclipse.lsp4j.debug.services.IDebugProtocolClient
+import org.eclipse.lsp4j.jsonrpc.services.JsonRequest
+import scala.concurrent.ExecutionContext
+import scala.meta.internal.debug.protocol.ExitedEventArguments
+import scala.meta.internal.debug.protocol.LaunchParameters
+
+final class JvmDebugAdapter(implicit val ex: ExecutionContext)
+    extends DebugAdapter {
+  private val client = new ClientDelegate
+  private val sessionFactory = new DebugSessionFactory(client)
+
+  @JsonRequest
```

Comment:
Are these annotations necessary on `JvmDebugAdapter`? I think it's only needed for `MetalsDebugServer`.

Reply:
If we want to maintain clear separation of concerns then yes, it is needed.
Otherwise `internal.eval` would have to be aware of the internals of`internal.metals`